### PR TITLE
Fix parseTime in Custom Parser and Formatter demo [skip ci]

### LINKED ISSUE #100: Example code for parseTime and formatTime is buggy
On the HTML examples page (https://vaadin.com/components/vaadin-time-picker/html-examples/time-picker-advanced-demos#custom-parser-and-formatter), the sample code for setting a new i18n object seems to be buggy. If you try to select 12:00 AM or 12:00 PM it bugs out on you and nothing gets selected. I decided to throw in console.logs to check what's going on. Here are my log statements (I put them right after the function calls):
```javascript
console.log("Calling formatTime", timeObject);
console.log("Calling parseTime ", timeString);
```
And here is the output for selecting 12:00 AM. When you select it, the time picker is still blank as if you haven't selected any time at all. It seems to freak out like it's 12:00 AM, then 12:00 PM, then 24:00, and then it doesn't know what to do with 24. 
Hopefully this is enough info and a fix can be pushed out soon. 

```
18:04:58.776 Calling parseTime  12:00 AM
18:04:58.777 Calling formatTime Object { hours: 12, minutes: "00" }
18:04:58.790 Calling parseTime  12:00 PM 
18:04:58.791 Calling formatTime Object { hours: 24, minutes: "00" }
18:04:58.794 Calling formatTime undefined 
18:04:58.800 Calling parseTime  
18:04:58.800 Calling formatTime undefined
18:04:58.817 Calling parseTime  12:00 AM
18:04:58.818 Calling formatTime Object { hours: 12, minutes: "00" }
18:04:58.821 Calling parseTime  12:00 PM 
18:04:58.821 Calling formatTime Object { hours: 24, minutes: "00" }
18:04:58.822 Calling formatTime undefined
18:04:58.828 Calling parseTime 
18:04:58.828 Calling formatTime undefined
18:04:58.833 Calling parseTime  12:00 AM
18:04:58.833 Calling formatTime Object { hours: 12, minutes: "00" }
18:04:58.837 Calling parseTime  12:00 PM 
18:04:58.837 Calling formatTime Object { hours: 24, minutes: "00" }
18:04:58.838 Calling formatTime undefined
18:04:58.840 Calling parseTime 
18:04:58.840 Calling formatTime undefined
```

### PR DESCRIPTION
**Pull request by shadikhani**
Fix https://github.com/vaadin/90min-dx-tasks/issues/111

Fixes #100

### CODE REVIEW

File: demo/time-picker-advanced-demos.html [hunk 1]
```diff
@@ -29,8 +29,21 @@ <h3>Custom Parser And Formatter</h3>
...
+
+                  if (parts) {
+                    if (parts[1] === '12' && parts[3] === 'PM') {
+                      // for 12pm noon, set hours to 12
+                      hours = 12;
+                    } else if (parts[1] === '12' && parts[3] === 'AM') {
+                      // for 12am midnight, set hours to 0
+                      hours = 0;
+                    } else {
+                      hours = parseInt(parts[1]) + (parts[3] === 'PM' ? 12 : 0);
```
Comment (samie) [inline demo/time-picker-advanced-demos.html]: Else statement could use similar explanation comment than the other if statements.

Comment (hongduc-phan) [reply demo/time-picker-advanced-demos.html]: hi, in my opinion, you can refactor if statements like this:
if (parts) {
    if (part[1] === '12') {
         if (part[3] === 'PM') {
              hours = 12;
         }
          else if (part[3] ==='AM') {
               hours = 0;
        }
    } 
   else {
          hours = parseInt(parts[1]) + (parts[3] === 'PM' ? 12 : 0);
  }
}

just for my opinion.

Comment (shadikhani) [reply demo/time-picker-advanced-demos.html]: Done.

Comment (shadikhani) [reply demo/time-picker-advanced-demos.html]: @hongduc-phan  Done.

Comment (hongduc-phan) [reply demo/time-picker-advanced-demos.html]: thank you

File: demo/time-picker-advanced-demos.html [hunk 2]
```diff
@@ -43,11 +59,12 @@ <h3>Custom Parser And Formatter</h3>
...
     <h3>Custom Validator</h3>
     <p>Extend <code>Vaadin.TimePickerElement</code> to create your own custom element,
-       then override the <code>checkValidity()</code> method to validate the input.</p>
+      then override the <code>checkValidity()</code> method to validate the input.
+    </p>
     <vaadin-demo-snippet id="time-picker-advanced-demos-custom-validator" when-defined="vaadin-time-picker" ignore-ie>
       <template preserve-content>
         <vaadin-time-picker-working-hours label="Delivery hour"
-          error-message="Select a vaild working hour (9:00 to 17:00)"></vaadin-time-picker-working-hours>
+        error-message="Select a vaild working hour (9:00 to 17:00)"></vaadin-time-picker-working-hours>
```
Comment (tmattsso) [inline demo/time-picker-advanced-demos.html]: typo :)

Comment (shadikhani) [reply demo/time-picker-advanced-demos.html]: Done

**Pull request merged**